tests/integration_tests/test_gas.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+
+from .utils import (
+    ADDRS,
+    CONTRACTS,
+    KEYS,
+    deploy_contract,
+    send_transaction,
+    w3_wait_for_new_blocks,
+)
+
+
+def test_gas_eth_tx(geth, evmos):
+    tx_value = 10
+
+    # send a transaction with geth
+    geth_gas_price = geth.w3.eth.gas_price
+    tx = {"to": ADDRS["community"], "value": tx_value, "gasPrice": geth_gas_price}
+    geth_receipt = send_transaction(geth.w3, tx, KEYS["validator"])
+
+    # send an equivalent transaction with evmos
+    evmos_gas_price = evmos.w3.eth.gas_price
+    tx = {"to": ADDRS["community"], "value": tx_value, "gasPrice": evmos_gas_price}
+    evmos_receipt = send_transaction(evmos.w3, tx, KEYS["validator"])
+
+    # ensure that the gasUsed is equivalent
+    assert geth_receipt.gasUsed == evmos_receipt.gasUsed
+
+
+def test_gas_deployment(geth, evmos):
+    # deploy an identical contract on geth and evmos
+    # ensure that the gasUsed is equivalent
+    _, geth_contract_receipt = deploy_contract(geth.w3, CONTRACTS["TestERC20A"])
+    _, evmos_contract_receipt = deploy_contract(evmos.w3, CONTRACTS["TestERC20A"])
+    assert geth_contract_receipt.gasUsed == evmos_contract_receipt.gasUsed
+
+
+def test_gas_call(geth, evmos):
+    function_input = 10
+
+    # deploy an identical contract on geth and evmos
+    # ensure that the contract has a function which consumes non-trivial gas
+    geth_contract, _ = deploy_contract(geth.w3, CONTRACTS["BurnGas"])
+    evmos_contract, _ = deploy_contract(evmos.w3, CONTRACTS["BurnGas"])
+
```

Comment:
I don't know if it could possibly make a difference to check calling this for the precompiles? Specifically something that changes the cosmos state? could be good to add as another tests

Reply:
Great suggestion @MalteHerrmann!! Will create a ticket to include this on a different PR